contracts/TxRelay.sol

```diff
@@ -63,4 +69,44 @@ contract TxRelay {
     function getNonce(address add) public constant returns (uint) {
         return nonce[add];
     }
+
+    /*
+     * @dev Adds a number of addresses to a specific whitelist. Only
+     * the owner of a whitelist can add to it.
+     * @param listNum The list number
+     * @param sendersToUpdate the addresses to add to the whitelist
+     */
+    function addToWhitelist(uint listNum, address[] sendersToUpdate) public {
+        if (listOwner[listNum] == 0x0) {
+            listOwner[listNum] = msg.sender;
+        }
+        updateWhitelist(listNum, sendersToUpdate, true);
+    }
+
+    /*
+     * @dev Removes a number of addresses from a specific whitelist. Only
+     * the owner of a whitelist can remove from it.
+     * @param listNum The list number
+     * @param sendersToUpdate the addresses to add to the whitelist
+     */
+    function removeFromWhitelist(uint listNum, address[] sendersToUpdate) public {
+        updateWhitelist(listNum, sendersToUpdate, false);
+    }
+
+    /*
+     * @dev Internal logic to update a whitelist
+     * @param listNum The list number
```

Comment:
So everyone basically picks their own list number? Seems that can lead to some weird cases where you’d have to hunt around for a list number that’s not already taken? And everyone needs to remember what their list number is. Not sure what the right approach is, maybe use claimedsender for namespace instead of listnum?

Reply:
You could iterate through all numbers using the public listOwner mapping. But what you are saying is that we change the whitelist mapping to be:
```
mapping(address => mapping(address => bool)) public whitelist;
```
Then using address`0x0` would be allow everything instead of `0`. 
I think this would be a bit more elegant.

Reply:
@oed  Yeah I think using that mapping would be simpler. The only issue is if for some reason you'd want to have multiple whitelists for a single sender, but that might not be relevant.

Reply:
@christianlundkvist Sounds good, and it should be simple enough to create a new address for new whitelists.